src/osd/OSD.cc

```diff
@@ -7657,7 +7662,14 @@ vector<DaemonHealthMetric> OSD::get_health_metrics()
            << " currently "
            << op.state_string();
         lgeneric_subdout(cct,osd,20) << ss.str() << dendl;
-        clog->warn() << ss.str();
+        if (log_slowop) {
+          clog->warn() << ss.str();
+        } else {
+          std::string_view state = op.state_string();
+          uint64_t const *poolid = op.get_poolid();
+          slow_types[state]++;
```

Comment:
I really really do not like the way slow_types is indexed:
we have the OP's latest_flag_point, which is a uint_8 with (currently) 7 possible values, and:
- we turn it into a string,
- we use the string as an index in a map - i.e. multiple string comparisons in a heavy-weight container.
and all that - when the original issue is performance.

At the minimum, I'd suggest:
- exposing both  latest_flag_point and its highest enum value (this one - as a constant expression) from the OP
  (I would have changed that to be a class enum, BTW)
- using that number as an index to an array of that size

Reply:
> I really really do not like the way slow_types is indexed: we have the OP's latest_flag_point, which is a uint_8 with (currently) 7 possible values, and:
> 
>     * we turn it into a string,
> 
>     * we use the string as an index in a map - i.e. multiple string comparisons in a heavy-weight container.
>       and all that - when the original issue is performance.
> 
> 
> At the minimum, I'd suggest:
> 
>     * exposing both  latest_flag_point and its highest enum value (this one - as a constant expression) from the OP
> 
>     * using that number as an index to an array of that size

@ronen-fr The latest_flag_point returns flag point related to slow op type for a tracked slow op. I am using latest_flag_point value as a index and incrementing counter against it. Does this sounds good to you ? Do you want me to map latest_flag_point 7 possible values to array index 0-6 ?